### ISSUE TITLE
InverseGeneralMapping regression, revert 59097e248?

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#944]
In 59097e248a1dc95f927eaf6ec1539fd4ce8c34e2 @hulpke changed the behavior of InverseGeneralMapping. Previously, the resulting generalized mapping simply had the source and range of the original map switched. But now, the source of the new generalized mapping is restricted to the *image* of the original map.

To quote my own comment on that change back then
```
@hulpke Hmm, I am worried about this change. For me, a generalized mapping 
is a possible non-total relation in R x S, and its generalized inverse is a relation 
in S x R. This change breaks that assumption.

Granted, the documentation of InverseGeneralMapping does not promise that, 
but I wonder if it maybe should, and instead of this change, we instead should 
fix any code that relies on a general mapping being total?
```

Well, I just realized that since then, the polycylic test suite has been broken, which tested explicitly for the old behaviour. Of course I can now change the polycyclic test suite accordingly... but my concerns are still there, and I don't think the change in 59097e248 are "correct" in that they are directly against the spirit of `InverseGeneralMapping`.

Perhaps we can add an `InverseMapping` which does what @hulpke wanted?

**Comment by hulpke** [comment]
The change you referred to was in response to a problem, though I can't find the details any more.

I think the problem is in part that there is no declaration of a ``practical inverse'' or ``total inverse'', restricted to the image.

I think a separate declaration of InverseMapping would be fine -- in this case most of my library code for groups would use this new operation.

How should these two operations be related? Could one refer to the other? For which of the two operations should the existing kernel methods work? (Do we need almost duplicate methods?)

**Comment by fingolfin** [comment]
It would be good to resolve this before 4.9, if possible, as this causes regressions compared to 4.8. i'll post more to tomorrow after some sleep.

**Comment by alex-konovalov** [comment]
The fix for this is in #1937.
